contracts/Advertisement.sol

```diff
@@ -231,123 +220,97 @@ contract Advertisement {
-			public view returns (uint){
+    function getTotalCampaignsByCountry (string country) public view returns (uint){
         bytes memory countryInBytes = bytes(country);
 
         return campaignsByCountry[countryInBytes].length;
     }
 
-    function getPackageNameOfCampaign (bytes32 bidId)
-			public view returns(string) {
-
-        return campaigns[bidId].filters.packageName;
+    function getPackageNameOfCampaign (bytes32 bidId) public view returns(string) {
+        return advertisementStorage.getCampaignPackageNameById(bidId);
     }
 
-    function getCountriesOfCampaign (bytes32 bidId)
-			public view returns(string){
-
-        return campaigns[bidId].filters.countries;
+    function getCountriesOfCampaign (bytes32 bidId) public view returns(string){
+        return advertisementStorage.getCampaignCountriesById(bidId);
     }
 
-    function getVercodesOfCampaign (bytes32 bidId)
-			public view returns(uint[]) {
-
-        return campaigns[bidId].filters.vercodes;
+    function getVercodesOfCampaign (bytes32 bidId) public view returns(uint[]) {
+        return advertisementStorage.getCampaignVercodesById(bidId);
     }
 
-    function getPriceOfCampaign (bytes32 bidId)
-			public view returns(uint) {
-
-        return campaigns[bidId].price;
+    function getPriceOfCampaign (bytes32 bidId) public view returns(uint) {
+        return advertisementStorage.getCampaignPriceById(bidId);
     }
 
-    function getStartDateOfCampaign (bytes32 bidId)
-			public view returns(uint) {
-
-        return campaigns[bidId].startDate;
+    function getStartDateOfCampaign (bytes32 bidId) public view returns(uint) {
+        return advertisementStorage.getCampaignStartDateById(bidId);
     }
 
-    function getEndDateOfCampaign (bytes32 bidId)
-			public view returns(uint) {
-
-        return campaigns[bidId].endDate;
+    function getEndDateOfCampaign (bytes32 bidId) public view returns(uint) {
+        return advertisementStorage.getCampaignEndDateById(bidId);
     }
 
-    function getBudgetOfCampaign (bytes32 bidId)
-			public view returns(uint) {
-
-        return campaigns[bidId].budget;
+    function getBudgetOfCampaign (bytes32 bidId) public view returns(uint) {
+        return advertisementStorage.getCampaignBudgetById(bidId);
     }
 
-    function getOwnerOfCampaign (bytes32 bidId)
-			public view returns(address) {
-
-        return campaigns[bidId].owner;
+    function getOwnerOfCampaign (bytes32 bidId) public view returns(address) {
+        return advertisementStorage.getCampaignOwnerById(bidId);
     }
 
-    function getBidIdList ()
-			public view returns(bytes32[]) {
+    function getBidIdList() public view returns(bytes32[]) {
         return bidIdList;
     }
 
     function isCampaignValid(bytes32 bidId) public view returns(bool) {
-        Campaign storage campaign = campaigns[bidId];
+        var (,,,startDate, endDate, valid,,,,,) = advertisementStorage.getCampaign(bidId);
```

Comment:
I get it, that it might be more gas efficient but wouldn't it be more readable if we called startDate, EndDate and Validity by their get functions?

Reply:
I will change it.